src/factorizations/svd.jl

```diff
@@ -0,0 +1,206 @@
+  return private ? (:V, fieldnames(typeof(F))...) : (:U, :S, :V, :Vt)
+end
+
+Base.size(A::SVD, dim::Integer) = dim == 1 ? size(A.U, dim) : size(A.Vt, dim)
+Base.size(A::SVD) = (size(A, 1), size(A, 2))
+
+function Base.show(io::IO, mime::MIME{Symbol("text/plain")}, F::SVD)
+  summary(io, F)
+  println(io)
+  println(io, "U factor:")
+  show(io, mime, F.U)
+  println(io, "\nsingular values:")
+  show(io, mime, F.S)
+  println(io, "\nVt factor:")
+  return show(io, mime, F.Vt)
+end
+
+Base.adjoint(usv::SVD) = SVD(adjoint(usv.Vt), usv.S, adjoint(usv.U))
+Base.transpose(usv::SVD) = SVD(transpose(usv.Vt), usv.S, transpose(usv.U))
+
+# Conversion
+Base.AbstractMatrix(F::SVD) = (F.U * Diagonal(F.S)) * F.Vt
+Base.AbstractArray(F::SVD) = AbstractMatrix(F)
+Base.Matrix(F::SVD) = Array(AbstractArray(F))
+Base.Array(F::SVD) = Matrix(F)
+SVD(usv::SVD) = usv
+SVD(usv::LinearAlgebra.SVD) = SVD(usv.U, usv.S, usv.Vt)
+
+# functions default to LinearAlgebra
+# ----------------------------------
+"""
+    svd!(A; full::Bool = false, alg::Algorithm = default_svd_alg(A)) -> SVD
+
+`svd!` is the same as [`svd`](@ref), but saves space by
+overwriting the input `A`, instead of creating a copy. See documentation of [`svd`](@ref) for details.
+"""
+svd!(A; kwargs...) = SVD(LinearAlgebra.svd!(A; kwargs...))
+
+"""
+    svd(A; full::Bool = false, alg::Algorithm = default_svd_alg(A)) -> SVD
+
+Compute the singular value decomposition (SVD) of `A` and return an `SVD` object.
+
+`U`, `S`, `V` and `Vt` can be obtained from the factorization `F` with `F.U`,
+`F.S`, `F.V` and `F.Vt`, such that `A = U * Diagonal(S) * Vt`.
+The algorithm produces `Vt` and hence `Vt` is more efficient to extract than `V`.
+The singular values in `S` are sorted in descending order.
+
+Iterating the decomposition produces the components `U`, `S`, and `V`.
+
+If `full = false` (default), a "thin" SVD is returned. For an ``M
+\\times N`` matrix `A`, in the full factorization `U` is ``M \\times M``
+and `V` is ``N \\times N``, while in the thin factorization `U` is ``M
+\\times K`` and `V` is ``N \\times K``, where ``K = \\min(M,N)`` is the
+number of singular values.
+
+`alg` specifies which algorithm and LAPACK method to use for SVD:
+- `alg = DivideAndConquer()` (default): Calls `LAPACK.gesdd!`.
+- `alg = QRIteration()`: Calls `LAPACK.gesvd!` (typically slower but more accurate) .
+
+!!! compat "Julia 1.3"
+    The `alg` keyword argument requires Julia 1.3 or later.
+
+# Examples
+```jldoctest
+julia> A = rand(4,3);
+
+julia> F = BlockSparseArrays.svd(A); # Store the Factorization Object
+
+julia> A ≈ F.U * Diagonal(F.S) * F.Vt
+true
+
+julia> U, S, V = F; # destructuring via iteration
+
+julia> A ≈ U * Diagonal(S) * V'
+true
+
+julia> Uonly, = BlockSparseArrays.svd(A); # Store U only
+
+julia> Uonly == U
+true
+```
+"""
+svd(A; kwargs...) =
+  SVD(svd!(eigencopy_oftype(A, LinearAlgebra.eigtype(eltype(A))); kwargs...))
+
+LinearAlgebra.svdvals(usv::SVD{<:Any,T}) where {T} = (usv.S)::Vector{T}
```

Comment:
```suggestion
LinearAlgebra.svdvals(usv::SVD{<:Any,T}) where {T} = (usv.S)::AbstractVector{T}
```
to be more general, unless there is a reason to force it to be `Vector`?